### PR TITLE
PR - Root certificate authority requires escape characters on windows OS

### PR DESCRIPTION
**Pull request by gpltaylor**
# Description
Resolve windows path being escaped by helm pkg

## Type of change
- This pull request fixes a bug in Radius and has an approved issue (issue link required).

## Additional Description
Helm.runesUntil strips \\ (x92) rune from string when converting Buffer RuneReader to []runes. This issues is outside this package. Alternatives is to import this func and remove this or mutate path before sending to parser. Care also has to be take to support users who are using "c:\\\\dir\\\\file.crt" as a hack. This only works as Helm's runesUntil strips \ by using the next rune!

fixes: Root certificate authority requires escape characters on windows OS #6946

### CODE REVIEW

File: pkg/cli/helm/radiusclient.go [hunk 1]
```diff
@@ -141,6 +141,13 @@ func AddRadiusValues(helmChart *chart.Chart, options *RadiusOptions) error {
 	}
 
 	for _, arg := range options.SetFileArgs {
+		// Continue to support: c:\\testdir\\testfile.crt
+		// and convert: c:\testdir\testfile.crt to c:/testdir/testfile.crt
+
+		if !strings.Contains(arg, "\\\\") && strings.Contains(arg, "\\") {
```
Comment (rynowak) [inline pkg/cli/helm/radiusclient.go:147]: Should we check for windows first? `\` has a different meaning in linux paths. It's a pretty rare case but `\` is legal as part of a filename in linux.

Comment (rynowak) [reply pkg/cli/helm/radiusclient.go:147]: Even better we could check [os.PathSeparator](https://cs.opensource.google/go/go/+/refs/tags/go1.22.2:src/os/path_unix.go;l=10)

Comment (gpltaylor) [reply pkg/cli/helm/radiusclient.go:147]: How about adding an additional check to ` runtime.GOOS`.
This along with the `os.PathSeparator` to stop using magic strings in the search.

There is precedence of this being used within the project

https://github.com/radius-project/radius/blob/74a3f25de96c0cee2b689fe33c6f3499e725c02b/pkg/cli/tools/util.go#L24

Comment (gpltaylor) [reply pkg/cli/helm/radiusclient.go]: @rynowak clean up the logic... tested on both Windows and Linux.

image

Comment (rynowak) [reply pkg/cli/helm/radiusclient.go]: My first time seeing [filepath.ToSlash](https://pkg.go.dev/path/filepath#ToSlash).

Fix looks great!

**Comment by rynowak** [comment]
@gpltaylor - thanks! the code change looks good. I've kicked off a functional test run, and I'll take care of merging when it passes.

**Comment by rynowak** [comment]
Note: the codecoverage check that's failing is new and we're stilling adjusting it. Right now it fails on small changes like this one, so I'll need to bypass the check.

**Comment by rynowak** [comment]
Merged! thanks

**Pull request merged**